src/test/java/pl/gemiusz/Case0022SetOrRefreshTokenSimulation.java

```diff
@@ -46,41 +48,59 @@ public class Case0022SetOrRefreshTokenSimulation extends Simulation {
                     .exec(
                             http("GeMi_SomeRequestWithToken_get")
-                                    .get("/get?foo=#{tokenStringSession}")
+                                    .get("/get")
+                                    .queryParam("foo", "#{tokenStringSession}")
                                     .check(jmesPath("args.foo").saveAs("tokenString4Print"))
                                     .check(jmesPath("args.foo").not("NOT_SET"))
-                                    //.check(jmesPath("args.foo").is(session -> tokenString))
+                            //.check(jmesPath("args.foo").is(session -> tokenString))
                     ).exec(session -> {
                         System.out.println("----------------------------------------------------------");
                         System.out.println("GeMi_SomeRequestWithToken_tokenString: " + tokenString);
-                        System.out.println("GeMi_SomeRequestWithToken_tokenStringSession: " + session.get("tokenStringSession").toString());
-                        System.out.println("GeMi_SomeRequestWithToken_tokenString4Print: " + session.get("tokenString4Print").toString());
+                        System.out.println("GeMi_SomeRequestWithToken_tokenStringSession: " + session.getString("tokenStringSession"));
+                        System.out.println("GeMi_SomeRequestWithToken_tokenString4Print: " + session.getString("tokenString4Print"));
                         System.out.println("----------------------------------------------------------");
                         return session;
                     });
 
 
     {
-        setUp(
-                scnForSetOrRefreshToken.injectOpen(
-                        //Duration of this scenario = 1. Time between requests * (2. Number of requests - 1)
-                        Stream.generate(
-                                () -> new OpenInjectionStep[]{
-                                        atOnceUsers(1),
-                                        nothingFor(Duration.ofSeconds(10)) //1. Time between requests
-                                }
-
-                        ).limit(4) //2. Number of requests
-                                .flatMap(Stream::of).toArray(OpenInjectionStep[]::new)),
-                scn.injectOpen(
-                        nothingFor(Duration.ofSeconds(2)),
-                        constantUsersPerSec(20).during(30)
+        setUp(scnForSetFirstToken.injectOpen(atOnceUsers(1))
+                .andThen(
+                        scnForRefreshToken.injectOpen(atOnceUsers(1)),
+
+                        //--------------------------------
+                        scn.injectOpen(
+                                constantUsersPerSec(10).during(30)
+                                //--------------------------------
+
+                        ).andThen(scnFinishTokenGeneration.injectOpen(atOnceUsers(1)))
```

Comment:
I think you will be able to extract the token management to be able to call like that:

```suggestion
        setUp(withTokenManagement(
                        scn.injectOpen(
                                constantUsersPerSec(10).during(30)))
```

WDYT?

Reply:
I was thinking about your suggestion to wrap Token Managment into some method and I'm not sure how to prepare it correctly and how to use it in `setUp`, some POC code below:
```java
    PopulationBuilder withTokenManagement(PopulationBuilder mainPopulation) {
        return scnForSetFirstToken.injectOpen(atOnceUsers(1))
                .andThen(
                scnForRefreshToken.injectOpen(atOnceUsers(1)),
                        mainPopulation
        ).andThen(scnFinishTokenGeneration.injectOpen(atOnceUsers(1)));
    }
```
any code examples will be nice :)

Reply:
Good idea to extract things about Toket to another Class and make it as library.
At this project adding Token Management will be overkilled but I will have this in my mind for other.

Thank you for help and suggestions.